airbyte-webapp/src/views/Connection/ConnectionForm/ConnectionForm.tsx

```diff
@@ -197,6 +197,35 @@ const ConnectionForm: React.FC<ConnectionFormProps> = ({
       {({ isSubmitting, setFieldValue, isValid, dirty, resetForm, values }) => (
         <FormContainer className={className}>
           <FormChangeTracker changed={dirty} formId={formId} />
+          <Section title={null}>
+            <Field name="name">
+              {({ field, meta }: FieldProps<string>) => (
+                <FlexRow>
+                  <LeftFieldCol>
+                    <ConnectorLabel
+                      nextLine
+                      error={!!meta.error && meta.touched}
+                      label={formatMessage({
+                        id: "form.connectionName",
+                      })}
+                      message={formatMessage({
+                        id: "form.connectionName.message",
+                      })}
+                    />
+                  </LeftFieldCol>
+                  <RightFieldCol style={{ pointerEvents: mode === "readonly" ? "none" : "auto" }}>
+                    <Input
+                      {...field}
```

Comment:
I would try out this. I haven't tested it but it should work. The reason why the style is changed below is that the dropdowns are not easily disabled.

```suggestion
                  <RightFieldCol>
                    <Input
                      {...field}
                      disabled={mode === "readonly"}
```